src/Routing/Router.php

```diff
@@ -979,7 +982,39 @@ public static function scope($path, $params = [], $callback = null)
         if (is_array($params)) {
             $options = $params;
             unset($params['routeClass'], $params['extensions']);
+            $cache = (string)Hash::get($params, '_cache');
+            if (!empty($cache)) {
+                unset($params['_cache']);
+                static::$_collection = Cache::remember(
+                    $path,
+                    function () use ($path, $params, $options, $callback) {
+                        self::_buildScope($path, $params, $options, $callback);
+
+                        return static::$_collection;
```

Comment:
I'm now checking for serializable, if not, I'm logging a warning message. I don't think we should handle the serialization in the Router as I was about to do. Throwing an exception is an option too, but maybe too hard and could break apps when going live badly if the cache is not correctly configured...